### PR TITLE
Sport for git 4

### PR DESCRIPTION
**Pull request by AnatoliyDubovtsev**
I added possibility to buy tickets for sport events and making requests to the city administration. There were some problems in the <date>_Fligth_Last migration, so I commented out 2 migration builders

### CODE REVIEW

File: UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            _mapper = mapper;
+        }
+
+        public IActionResult AllRequestsToAdministration()
+        {
+            List<RequestToAdministration> allRequests = _requestToAdministrationService.GetAllRequestsToAdministration();
+            List<RequestToAdministrationViewModel> viewModels = _requestToAdministrationService.CreatingRequestToAdministationViewModel(allRequests, _mapper);
+            ViewBag.SportComplexesIds = _sportComplexService.GetAllSportComplexesIds();
+            ViewBag.IsMainPage = true;
+            return View(viewModels);
```
Comment (crocomoth) [inline UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs]: Would be good if name was specified

File: UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        }
+
+        public IActionResult AllRequestsToAdministrationByDate(DateTime date)
+        {
+            List<RequestToAdministration> allRequests = _requestToAdministrationService.GetRequestsToAdministrationByDate(date);
```
Comment (crocomoth) [inline UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs]: probably can be `var` - it is more or less clear what type it is, also temporary variable

File: UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                return View("Error", "Some errors in input data. Please, try again");
+            }
+
+            SportComplex sportComplex = _sportComplexService.GetSportComplexByTitle(requestViewModel.SportComplexTitle);
+            if (sportComplex == null)
+            {
+                return View("Error", "Some problems with the sport complex data. Please, try again");
+            }
+
+            if (requestViewModel.IsApproved && !sportComplex.Available)
```
Comment (crocomoth) [inline UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs]: probably sportcomplex.available can be ignored and just overwritten based on requestViewModel. You are updating db anyway, so overwriting bool property should not be a problem really

File: UtopiaCity/Controllers/Sport/SportTicketController.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        public SportTicketController(SportTicketService sportTicketService, SportComplexService sportComplexService, SportEventService sportEventService,
+            CitizensAccountService appUserService, CitizenTaskService citizenTaskService, IMapper mapper, IHttpContextAccessor httpContextAccessor)
+        {
+            _sportTicketService = sportTicketService;
+            _sportComplexService = sportComplexService;
+            _sportEventService = sportEventService;
+            _appUserService = appUserService;
+            _citizenTaskService = citizenTaskService;
+            _mapper = mapper;
+            _userId = httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
```
Comment (crocomoth) [inline UtopiaCity/Controllers/Sport/SportTicketController.cs]: Better to wrap this logic - can be changed any time - better to have 1 class to do 1 thing, being SOLID is cool

File: UtopiaCity/Data/Migrations/AirportMigrations/20210808153541_Flight_Last.cs [hunk 5]
```diff
@@ -6,9 +6,9 @@ public partial class Flight_Last : Migration
     {
         protected override void Up(MigrationBuilder migrationBuilder)
         {
-            migrationBuilder.DropColumn(
-                name: "TravelDistance",
-                table: "Flights");
+//            migrationBuilder.DropColumn(
```
Comment (crocomoth) [inline UtopiaCity/Data/Migrations/AirportMigrations/20210808153541_Flight_Last.cs]: ?

File: UtopiaCityTest/Common/ObjectsForTests/SportObjectsForTests.cs [hunk 6]
```diff
@@ -130,5 +133,165 @@ public static SportEvent[] ArrayOfSportEventsForTests()
...
+				    DateOfBirth = new DateTime(2001, 1, 1),
+				    Gender = Gender.Male,
+                    Balance = 1.1
+				},
+
+				new AppUser
+				{
+					Id = "2",
+				    Name = "Name_2",
+                    Surname = "Surname_2",
```
Comment (crocomoth) [inline UtopiaCityTest/Common/ObjectsForTests/SportObjectsForTests.cs:225]: intendation

**Review by crocomoth — APPROVED** [review]

File: UtopiaCity/Data/Migrations/AirportMigrations/20210808153541_Flight_Last.cs (hunk 5 above)
Comment (AnatoliyDubovtsev) [reply UtopiaCity/Data/Migrations/AirportMigrations/20210808153541_Flight_Last.cs]: There were problems in time of using the "update-database" command so I commented out it

**Review by IvanSultanov — APPROVED** [review]

**Pull request merged**